### DISCUSSION TITLE
Updating/Reassigning Object in Array

### DISCUSSION THREAD

**Discussion by MatthewMahabeer** [pmndrs/jotai#1277] [Q&A]
Saw many examples of mutating objects and arrays but not an array of objects and the conventional methods don't seem to be working. Any help? Simply trying to find and update an object in the array.

`
<img width="560" alt="Screenshot 2022-07-10 162042" src="https://user-images.githubusercontent.com/94992417/178162388-f632fafe-f8c8-4275-b8d5-895c13c135a0.png">

Instead of Object.assign, I also tried deckList[index] = get(deckOperatorAtom).

**Comment by dai-shi** [comment]
Updating arrays immutably is not very easy. This applies to not only jotai but `useState` and others too.
You could do in various ways, including libraries, but what I would usually do in plain JS is this:

```js
const arr = [{ id: 1, name: 'React' }, { id: 2, name: 'Hooks' }]

// updating name of id=1 to "Jotai"
const index = arr.findIndex((item) => item.id === 1)
const newArr = [
  ...arr.slice(0, index),
  { ...arr[index], name: 'Jotai' },
  ...arr.slice(index + 1)
]
```

**Reply by MatthewMahabeer** [reply]
Thanks for the reply!
Will that approach work within the (get, set) function in jotai?

**Reply by dai-shi** [reply]
Yes, it should.

Copy-and-mutate should work too:
```js
const index = arr.findIndex((item) => item.id === 1)
const newArr = [...arr]
newArr[index] = { ...arr[index], name: 'Jotai' }
```
